src/auto_contiguous.cpp

```diff
@@ -64,7 +64,10 @@ void auto_contiguous::apply(module& m) const
         if(ins->outputs().empty() and ins != last)
             continue;
         shape s = ins->get_shape();
-        if(not s.dynamic() and not s.standard() and s.elements() != 0)
+        // If s is not standard layout or has out of sequence strides, insert "contiguous" op
+        // to make a standard shape
+        if(not s.dynamic() and (not s.standard() or s.normalize_standard() != s) and
+            s.elements() != 0)
```

Comment:
This needs to be `s.elements() > 1` to fix the tests.

Reply:
I made that comment after having made and tested that change on my machine but I didn't push it.  I just pushed the change so you can see for yourself that it still fails.